hs_file_types/models/base.py

```diff
@@ -638,6 +639,11 @@ class AbstractLogicalFile(models.Model):
     # also this data type needs to be defined in in terms.html page
     data_type = "Generic"
 
+    # this field is for logical file to store extra key:value pairs, e.g., currently for .url
+    # file to store url value for easy redirection when opening the file
+    # for internal use only - won't get recorded in bag and shouldn't be used for storing metadata
+    extra_data = HStoreField(default={})
```

Comment:
Should this only be on the SingleFile instead of all types?

Reply:
And a follow up.  Is there any reason we can't just reserve a key in the extra_metadata instead of adding a new field?

Reply:
@sblack-usu The reason I did not use extra_metadata for referenced url is that extra_metadata is part of file metadata and is editable by users from UI and also show up in file metadata xml file. We have extra_data in baseresoruce model, reserved only for system use, so I thought it is a good idea to replicate it to file level, not only for referenced url key for this functionality, but also  can be potentially used for other purposes in the future. I remember I recorded this in my code comment when defining this extra_data field in the genericlogicalfile model.

Reply:
@sblack-usu I added this on the base model on purpose so all logical file types will get this field. The reason is that I see this extra_data field as a generically applicable system-only field that could come in handy for other logical file types in the future although it is only used for referenced url on SingleFile type.